src/pretix/api/views/event.py

```diff
@@ -240,6 +241,42 @@ def get_queryset(self):
             'subeventitem_set', 'subeventitemvariation_set'
         )
 
+    def perform_update(self, serializer):
+        super().perform_update(serializer)
+
+        serializer.instance.log_action(
+            'pretix.subevent.changed',
+            user=self.request.user,
+            auth=self.request.auth,
+            data=self.request.data
+        )
+
+    def perform_create(self, serializer):
+        serializer.save(event=self.request.event)
+        serializer.instance.log_action(
+            'pretix.subevent.added',
+            user=self.request.user,
+            auth=self.request.auth,
+            data=self.request.data
+        )
+
+    def perform_destroy(self, instance):
+        if not instance.allow_delete():
+            raise PermissionDenied('The event can not be deleted as it already contains orders. Please set \'active\''
+                                   ' to false to hide the event and take the shop offline instead.')
```

Comment:
I think this is still unclear in terms of terminology, how about ``The subevent can not be deleted as it has already been used in orders. Please set \'active\' to false instead to hide it from users.``